exec/java-exec/src/main/java/org/apache/drill/exec/client/DrillClient.java

```diff
@@ -223,19 +223,65 @@ public void connect(Properties props) throws RpcException {
     connect(null, props);
   }
 
+  /**
+   * Function to populate the endpointList with information of all the drillbits
+   * provided in the connection string by client
+   * @param endpointList - ArrayList of DrillbitEndpoints
+   * @param drillbits - One or more drillbit ip[:port] provided in connection string
+   */
+  public void populateEndpointsList(ArrayList<DrillbitEndpoint> endpointList, String drillbits){
+
+    // If no information about drillbits is provided then just return empty list.
+    if(drillbits == null || drillbits.length() == 0){
```

Comment:
drillbits.length( ) == 0 --> drillbits.trim( ).isEmpty( )

The trim removes any bogus white space. IsEmpty( ) is a shorthand for length( ) == 0